source/extensions/filters/http/jwt_authn/jwks_cache.cc

```diff
@@ -0,0 +1,169 @@
+#include "extensions/filters/http/jwt_authn/jwks_cache.h"
+
+#include <chrono>
+#include <unordered_map>
+
+#include "common/common/logger.h"
+#include "common/config/datasource.h"
+
+using ::google::jwt_verify::Jwks;
+using ::google::jwt_verify::Status;
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace JwtAuthn {
+namespace {
+
+// Default cache expiration time in 5 minutes.
+const int kPubkeyCacheExpirationSec = 600;
+
+// HTTP Protocol scheme prefix in JWT aud claim.
+const std::string kHTTPSchemePrefix("http://");
```

Comment:
Envoy doesn't allow static non-POD (like Google3), consider using `ConstSingleton` or the like.